src/game.cpp

```diff
@@ -13886,3 +13888,21 @@ std::vector<npc *> game::allies()
     } );
     return res;
 }
+
+void game::mon_lay_egg()
+{
+
+
+    for( size_t i = 0; i < num_zombies(); i++ ) {
+        monster &critter = critter_tracker->find( i );
+        const furn_t &furn = g->m.furn( critter.pos() ).obj();
+
+        if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
+            !critter.has_effect( effect_laid_egg ) ) {
+
+            item egg( "egg_bird", 0, 1 );
+            g->m.add_item_or_charges( critter.pos(), egg );
+            critter.add_effect( effect_laid_egg, ( HOURS( 48 ) ) );
+        }
+    }
```

Comment:
>Monster "attacks" don't need to be actual attacks, they're just abilities monsters use by themselves.

Didn't know that.

And what if I move mon_lay_egg() to monster::move() and add a check for egglaying flag and effect?